byway/src/main/webapp/track-interests.js

```diff
@@ -92,6 +93,22 @@ function createButtonForPlace(place) {
   return button;
 }
 
+/* Hide or display list of interests to match user input as a zero-indexed substring. */
+function filterInterests() {
+  const searchField = document.getElementById("search-interest");
+  const searchValue = searchField.value.toLowerCase();
+  const interestBtnsElement = document.getElementById("interests-buttons");
+  let interestBtns = interestBtnsElement.getElementsByClassName("interestBtn");
+  for (let i = 0; i < interestBtns.length; i++) {
+    const btnValue = interestBtns[i].innerText;
+    if (btnValue.toLowerCase().indexOf(searchValue) === 0) {
+      interestBtns[i].style.display = "";
```

Comment:
Seems weird to clear the display field instead of replacing it with another value to show that it's shown to the user. Try inline or inline-block.

Reply:
Fixed